src/osdc/Objecter.h

```diff
@@ -1645,6 +1645,10 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   std::atomic<bool> initialized{false};
 
 private:
+  struct fastfail {
+    std::set<pg_t> pgs;
```

Comment:
Don't use a standard set here. It's a tree structure and pretty slow and bulky for integers. Use `std::unordered_set` or, if you need to traverse them in order, `boost::container::flat_set`.

Reply:
done